File: phoenixfront/src/app/models/inventory/ProductDto.ts
import { AgentProdDto } from "./AgentProdDto";
import { State } from "./State";
import { Stockdto } from "./Stock";

export interface Productdto {
    serialNumber?: string;
    simNumber?: string;
    checkin?: Date;
    boxNumber?: string;
    brand?: string;
    productType?: string;
    prodName?: string;
    comments?: string;
    price?: number;
    state?: State;
    checkedExistence?: boolean;
    stock?: Stockdto;
    agentProd?: AgentProdDto;
    managerProd?: AgentProdDto;
     agentwhoSoldProd?: AgentProdDto;
     agentReturnedProd?: AgentProdDto;

}

File: phoenix/keycloakuser/src/main/java/com/phoenix/dtostock/ProductDto.java
package com.phoenix.dtostock;

import com.phoenix.model.State;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ProductDto {
    private String serialNumber;
    private String brand;
    private String productType;
    private String prodName;
    private String prodDescription;
    private BigDecimal price;
    private State state;
    private LocalDate soldDate;
    private boolean checked;
    private StockDto stock;
    private AgentProdDto agentProd;
}
